src/main/java/se/sundsvall/billingdatacollector/integration/opene/mapper/kundfakturaformular/KundfakturaformularMapper.java

```diff
@@ -0,0 +1,235 @@
+		var customerId = getLeadingDigits(result.forvaltningSomSkaBetala());
+
+		var billingRecord = BillingRecord.builder()
+			.withCategory(CATEGORY)
+			.withStatus(Status.APPROVED)
+			.withType(Type.INTERNAL)
+			.withRecipient(Recipient.builder()
+				.withOrganizationName(SUNDSVALLS_MUNICIPALIY)
+				.withLegalId(SUNSVALLS_MUNICIPALITY_ORGANIZATION_NUMBER)
+				.build())
+			.withInvoice(Invoice.builder()
+				.withCustomerId(customerId)
+				.withInvoiceRows(List.of(InvoiceRow.builder()
+					.withDescriptions(List.of(result.fakturaText()))
+					.withQuantity(result.antal())
+					.withCostPerUnit(convertStringToFloat(result.aPris()))
+					.withTotalAmount(convertStringToFloat(result.summering()))
+					.withAccountInformation(AccountInformation.builder()
+						.withCostCenter(getLeadingDigits(result.ansvar()))
+						.withSubaccount(getLeadingDigits(result.underkonto()))
+						.withDepartment(getLeadingDigits(result.verksamhet()))
+						.withActivity(getLeadingDigits(result.aktivitetskonto()))
+						.withCounterpart(getInternalMotpartNumbers(customerId))
+						.build())
+					.build()))
+				.build())
+			.build();
+
+		return BillingRecordWrapper.builder()
+			.withBillingRecord(billingRecord)
+			.build();
+	}
+
+	BillingRecordWrapper mapToExternalBillingRecord(final byte[] xml) {
+		var result = extractValue(xml, ExternFaktura.class);
+
+		var billingRecord = BillingRecord.builder()
+			.withCategory(CATEGORY)
+			.withStatus(Status.APPROVED)
+			.withType(Type.EXTERNAL)
+			.withRecipient(Recipient.builder()
+				.withFirstName(result.fornamn())
+				.withLastName(result.efternamn())
+				.withAddressDetails(AddressDetails.builder()
+					.withStreet(result.adress())
+					.withPostalCode(result.postnummer())
+					.withCity(result.ort())
+					.build())
+				.build())
+			.withInvoice(Invoice.builder()
+				.withInvoiceRows(List.of(InvoiceRow.builder()
+					.withDescriptions(List.of(result.fakturaText()))
+					.withQuantity(result.antal())
+					.withCostPerUnit(convertStringToFloat(result.aPris()))
+					.withVatCode(result.momssats())
+					.withTotalAmount(convertStringToFloat(result.summering()))
+					.withAccountInformation(AccountInformation.builder()
+						.withCostCenter(getLeadingDigits(result.ansvar()))
+						.withSubaccount(getLeadingDigits(result.underkonto()))
+						.withDepartment(getLeadingDigits(result.verksamhet()))
+						.withActivity(getLeadingDigits(result.aktivitetskonto()))
+						.withArticle(result.objektkonto())
+						.withCounterpart(getExternalMotpartNumbers(result.motpartNamn()))
+						.build())
+					.build()))
+				.build())
+			.build();
+
+		return BillingRecordWrapper.builder()
+			.withBillingRecord(billingRecord)
+			.withLegalId(result.personnummer())
+			.build();
+	}
+
+	//////
+	// Lots of converters and stuff down here.
+	// Move to a separate class if we need them in more places in the future.
```

Comment:
Instead of writing a comment about a separate class for this: Write a separate class for this. :-)
(some of the methods maybe is maybe even candidates for dept44 in some string util class?

Reply:
I'll make a separate class.